### PR TITLE
update Icon to TS version in UI Folder

### LINKED ISSUE #18492: DS/Extension/2023/Q2/O1/KR3: Icon import clean up
### Description
Continue `Icon` import clean up and migration of component to TS

### Tasks (separate PRs)
- [x] Clean up imports and update JS to TS `ui/components/multichain`
- [x] Clean up imports and update JS to TS `ui/components/institutional`
- [x] Clean up imports and update JS to TS `ui/components/ui`
- [ ] Clean up imports and update JS to TS `ui/components/app`
- [ ] Clean up imports and update JS to TS `ui/pages`
- [ ] Clean up imports in all other folders
- [ ] Delete `component-name/deprecated` folder with JS components and consts

### Technical Details
- Update imports from  `'../../component-library/component-name/deprecated';` to `'../../component-library'`
- Update component instances from using JS consts to enums

### Acceptance Criteria 
- All components have been migrated to TS

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

 UI folder Icon imports updated to TS version

Update imports from '../../component-library/component-name/deprecated'; to '../../component-library'
Update component instances from using JS consts to enums

* Fixes #18492 

## Screenshots/Screencaps

See comments

## Manual Testing Steps
`yarn jest ui/components/component-library`

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/ui/menu/menu.stories.js [hunk 1]
```diff
@@ -29,14 +29,11 @@ export const Anchored = () => {
...
-        <MenuItem iconName={ICON_NAMES.EXPORT} onClick={action('Menu Item 1')}>
+        <MenuItem iconName={IconName.Export} onClick={action('Menu Item 1')}>
           Menu Item 1
         </MenuItem>
         <MenuItem onClick={action('Menu Item 2')}>Menu Item 2</MenuItem>
-        <MenuItem
-          iconName={ICON_NAMES.EYE_SLSH}
-          onClick={action('Menu Item 3')}
-        >
+        <MenuItem iconName={IconName.Eye_Slsh} onClick={action('Menu Item 3')}>
```
Comment (NidhiKJha) [inline ui/components/ui/menu/menu.stories.js:36]:
```suggestion
        <MenuItem iconName={IconName.Eye_Slash} onClick={action('Menu Item 3')}>
```

File: ui/components/ui/nickname-popover/nickname-popover.component.js [hunk 2]
```diff
@@ -67,8 +63,8 @@ const NicknamePopover = ({
             title={copied ? t('copiedExclamation') : t('copyToClipboard')}
           >
             <ButtonIcon
-              iconName={copied ? ICON_NAMES.COPY_SUCCESS : ICON_NAMES.COPY}
-              size={ICON_SIZES.SM}
+              iconName={copied ? IconName.CopySuccess : IconName}
```
Comment (NidhiKJha) [inline ui/components/ui/nickname-popover/nickname-popover.component.js:66]:
```suggestion
              iconName={copied ? IconName.CopySuccess : IconName.Copy}
```

**Review by NidhiKJha — COMMENTED** [review]
Looking great! Just some small nit fix

**Review by NidhiKJha — APPROVED** [review]

**Review by georgewrmarshall — COMMENTED** [review]
LGTM! Not sure the manual testing steps listed in the description are a good measure to test these changes?
- checked out branch and searched for `ICON_NAMES`

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**